### PR TITLE
Support for company fields in tracking code pageview event

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/266
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5499
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added support for company fields in     mt('send', 'pageview');
Inspirated by SubmissionModel https://github.com/mautic/mautic/blob/fed5d81bda03e25ca0d77108ca07b05de854c12a/app/bundles/FormBundle/Model/SubmissionModel.php#L1017

This PR also added more priority to companyname fields like take companyname from companyemail.

https://github.com/kuzmany/mautic/blob/1c1b1822aef2bec3f9ddebe0df8b7e0d7204569e/app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php#L76

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1.  Mark company fields as Publicly updatable  
2.  Use 
 `   mt('send', 'pageview', {companyemail: 'mycompany@mycompany.com', companywebsite: 'https://mautic.com/'});`

More in docs https://mautic.org/docs/en/contacts/contact_monitoring.html  (part Tracking of custom parameters)

3. See fields wasn't update in tracking contact

#### Steps to test this PR:
1.  Repeat alls teps
2.  See company fields was updated in Mautic

### CODE REVIEW

File: app/bundles/PageBundle/Model/PageModel.php [hunk 1]
```diff
@@ -463,6 +474,20 @@ public function hitPage($page, Request $request, $code = '200', Lead $lead = nul
         // Get lead if required
         if (null == $lead) {
             $lead = $this->leadModel->getContactFromRequest($query, $this->trackByFingerprint);
+            // company
+            list($company, $leadAdded, $companyEntity) =  IdentifyCompanyHelper::identifyLeadsCompany($query, $lead, $this->companyModel);
+            if ($leadAdded) {
+                $lead->addCompanyChangeLogEntry('form', 'Identify Company', 'Lead added to the company, '.$company['companyname'], $company['id']);
+            } else {
+                $this->companyModel->setFieldValues($companyEntity, $query);
```
Comment (escopecz) [inline app/bundles/PageBundle/Model/PageModel.php]: I'm getting `Type error: Argument 1 passed to Mautic\LeadBundle\Model\CompanyModel::setFieldValues() must be an instance of Mautic\LeadBundle\Entity\Company, null given` on this line.

Comment (kuzmany) [reply app/bundles/PageBundle/Model/PageModel.php]: Got it, fixed.

**Comment by kuzmany** [comment]
@escopecz 
As I read from source code https://github.com/kuzmany/mautic/blob/1c1b1822aef2bec3f9ddebe0df8b7e0d7204569e/app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php#L74

Mautic try find company name from 
company
companyname
if not exist from domain name of email/company email
or if not exist  domain name of companywebsite 

This is right workflow?

**Comment by kuzmany** [comment]
@escopecz maybe we can remove email  from findCompany https://github.com/kuzmany/mautic/blob/1c1b1822aef2bec3f9ddebe0df8b7e0d7204569e/app/bundles/LeadBundle/Helper/IdentifyCompanyHelper.php#L74 

A lot of people use personal email like gmail, outlook, yahoo, seznam.cz and nobody from them work in that companies :)

**Comment by kuzmany** [comment]
@Maxell92 documentation added https://github.com/mautic/documentation/pull/266

**Comment by escopecz** [comment]
Removing email domain from that code would be a business decision. Lets have it as is for now.

What I was noting is that if you use the sample code you provided:
`mt('send', 'pageview', {companyemail: 'mycompany@mycompany.com', companywebsite: 'https://mautic.com/'});`

It actually doesn't do anything. It won't create new company. The information will be lost. So some users could be searching for an answer why. The answer is that they must provide `companyname` in there too. Then the company will be created with the email and website they provided.

**Comment by kuzmany** [comment]
@escopecz your code create company with companyname **mycompany**
I am going to check again

**Comment by escopecz** [comment]
Nope, it did not create mycompany company because it check DNS MX record. But that's not the point. If you send in the page hit request companyaddress alone Mautic will ignore it. It must contain company or companyname to store the rest of the company* values. That I think should be documented.

**Comment by kuzmany** [comment]
@escopecz docs update https://github.com/mautic/documentation/pull/266

**Pull request merged**